### PR TITLE
about issue #78 Add version command at the CLI

### PR DESCRIPTION
**Pull request by shazaum**
Only shows version at the moment, pipes are a future feature.

### CODE REVIEW

**Comment by diegobernardes** [comment]
@shazaum, I don't know why the CI is not working on your branch. Let me fix that before we can proceed, ok?

**Comment by shazaum** [comment]
Yes, ok

File: cmd/pipehub/main.go [hunk 1]
```diff
@@ -17,9 +18,16 @@ import (
 
 var done = make(chan os.Signal, 1)
 
+var (
+	gitCommit string
+	goVersion string
```
Comment (diegobernardes) [inline cmd/pipehub/main.go]: Here you could do something different: `runtime.Version()`.

File: cmd/pipehub/main.go [hunk 2]
```diff
@@ -135,3 +143,31 @@ func cmdGenerateRun(configPath, workspacePath *string) func(*cobra.Command, []st
...
+		Run:   cmdVersionRun(gitCommit, goVersion, built, version),
+	}
+
+	cmd.Flags().StringVarP(&version, "version", "", "", "show version")
+	return &cmd
+}
+
+func cmdVersionRun(gitCommit, goVersion, built, version string) func(*cobra.Command, []string) {
+	return func(cmd *cobra.Command, args []string) {
+
```
Comment (diegobernardes) [inline cmd/pipehub/main.go]: Actually, you don't need this empty line, could you remove it? Just to keep the same pattern as the rest of the code.

File: Makefile [hunk 3]
```diff
@@ -4,6 +4,10 @@ DOCKER_CI_VERSION  = 2
 CONFIG_PATH       ?= $(CURDIR)/cmd/pipehub/pipehub.hcl
 WORKSPACE_PATH     = $(CURDIR)
 RAWTAG             = $(shell git tag --points-at | head -n1 | cut -c2-)
+GIT_COMMIT         = $(shell git rev-list -1 HEAD)
+GO_VERSION         = $(shell go version | cut -d' ' -f3)
```
Comment (diegobernardes) [inline Makefile]: I think we don't need this line.

File: Makefile [hunk 4]
```diff
@@ -4,6 +4,10 @@ DOCKER_CI_VERSION  = 2
...
+BUILT              = $(shell date)
```
Comment (diegobernardes) [inline Makefile]: What you think of renaming this variable to `BUILT_AT`? I think it's more semantic.

File: cmd/pipehub/main.go [hunk 5]
```diff
@@ -135,3 +143,31 @@ func cmdGenerateRun(configPath, workspacePath *string) func(*cobra.Command, []st
...
+		// Observe that the third line has no trailing tab,
+		// so its final cell is not part of an aligned column.
+		const padding = 3
+		w := tabwriter.NewWriter(os.Stdout, 0, 0, padding, ' ', tabwriter.TabIndent)
+		fmt.Fprintln(w, "PipeHub:")
+		fmt.Fprintln(w, "    Version:\t", version)
```
Comment (diegobernardes) [inline cmd/pipehub/main.go]: A good thing here is on each `fmt.Fprintln` check if the value is empty or not, if it is, we could skip the whole line. What do you think of it?

**Review by diegobernardes — CHANGES_REQUESTED** [review]

File: cmd/pipehub/main.go (hunk 2 above)
Comment (shazaum) [reply cmd/pipehub/main.go]: Yes, ok, I removed this blank line.

File: Makefile (hunk 4 above)
Comment (shazaum) [reply Makefile]: yes, I agree

File: cmd/pipehub/main.go (hunk 1 above)
Comment (shazaum) [reply cmd/pipehub/main.go]: good idea

File: cmd/pipehub/main.go (hunk 5 above)
Comment (shazaum) [reply cmd/pipehub/main.go]: fixed :D

**Review by diegobernardes — APPROVED** [review]

**Pull request merged**